### ISSUE TITLE
Unable to start dSIPRouter. Python error

### ISSUE THREAD

**Issue by habibshahid** [dOpensource/dsiprouter#386]
On a fresh install on Debian 9 using the One Line Install script. I am unable to start the service. following is the error I get. Please advise.

root@ip-10-81-1-119:/home/admin# lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 9.13 (stretch)
Release:        9.13
Codename:       stretch

************** Error **************
root@ip-10-81-1-119:/home/admin# dsiprouter restart -debug
+ shift
+ ((  0 == 0  ))
+ STOP_DSIPROUTER=1
+ START_DSIPROUTER=1
+ ((  0 > 0  ))
+ RUN_COMMANDS=($(printf '%s\n' "${RUN_COMMANDS[@]}" | awk '!x[$0]++'))
++ printf '%s\n' stop start
++ awk '!x[$0]++'
+ for RUN_COMMAND in "${RUN_COMMANDS[@]}"
+ stop
+ local STOP_DSIPROUTER=1
+ local STOP_KAMAILIO=0
+ local STOP_RTPENGINE=0
+ updatePythonRuntimeSettings
+ ((  1 == 1  ))
+ setConfigAttrib DEBUG True /etc/dsiprouter/gui/settings.py
+ local NAME=DEBUG
+ local VALUE=True
+ local CONFIG_FILE=/etc/dsiprouter/gui/settings.py
+ ((  3 >= 4  ))
+ sed -i -r -e 's|DEBUG[ \t]*=[ \t]*.*|DEBUG = True|g' /etc/dsiprouter/gui/settings.py
+ ((  0 == 1  ))
+ ((  0 == 1  ))
+ ((  1 == 1  ))
+ '[' -e /etc/dsiprouter/.dsiprouterinstalled ']'
+ systemctl stop nginx
+ systemctl stop dsiprouter
+ systemctl is-active --quiet dsiprouter
+ pprint 'dSIPRouter was stopped'
+ [[ dSIPRouter was stopped == \-\n ]]
+ printf '%b%s%b\n' '\033[1;36m' 'dSIPRouter was stopped' '\033[39;49;00m'
dSIPRouter was stopped
+ for RUN_COMMAND in "${RUN_COMMANDS[@]}"
+ start
+ local START_DSIPROUTER=1
+ local START_KAMAILIO=0
+ local START_RTPENGINE=0
+ updatePythonRuntimeSettings
+ ((  1 == 1  ))
+ setConfigAttrib DEBUG True /etc/dsiprouter/gui/settings.py
+ local NAME=DEBUG
+ local VALUE=True
+ local CONFIG_FILE=/etc/dsiprouter/gui/settings.py
+ ((  3 >= 4  ))
+ sed -i -r -e 's|DEBUG[ \t]*=[ \t]*.*|DEBUG = True|g' /etc/dsiprouter/gui/settings.py
+ ((  0 == 1  ))
+ ((  0 == 1  ))
+ ((  1 == 1  ))
+ '[' -e /etc/dsiprouter/.dsiprouterinstalled ']'
+ '[' 1 -eq 1 ']'
+ systemctl start nginx
+ sudo -u dsiprouter -g dsiprouter /usr/bin/python3 /opt/dsiprouter/gui/dsiprouter.py
Traceback (most recent call last):
  File "/opt/dsiprouter/gui/dsiprouter.py", line 15, in <module>
    from flask_wtf.csrf import CSRFProtect
  File "/usr/local/lib/python3.5/dist-packages/flask_wtf/__init__.py", line 1, in <module>
    from .csrf import CSRFProtect, CsrfProtect
  File "/usr/local/lib/python3.5/dist-packages/flask_wtf/csrf.py", line 220
    dest = f'{view.__module__}.{view.__name__}'
                                              ^
SyntaxError: invalid syntax
+ PID=
+ ps -p
+ printerr 'Unable to start dSIPRouter'
+ [[ Unable to start dSIPRouter == \-\n ]]
+ printf '%b%s%b\n' '\033[1;31m' 'Unable to start dSIPRouter' '\033[39;49;00m'
Unable to start dSIPRouter
+ cleanupAndExit 1
+ rm -f /etc/apt/apt.conf.d/local
+ set +x

**Comment by TotallyAProgrammer** [comment]
I'm experiencing this same issue, does has there been any progress on a fix?

**Comment by TotallyAProgrammer** [comment]
Okay, I figured it out. As it turns out, f-string was added in python 3.6, and is not available in python 3.5. I am going to create a patch and test it, I will report back my findings

**Comment by TotallyAProgrammer** [comment]
Here is the patch for the file /usr/local/lib/python3.5/dist-packages/flask_wtf/csrf.py

This can also be solved by simply upgrading your python version to version >= 3.6


Line 220:
```
#dest = f'{view.__module__}.{view.__name__}'
dest = '{}.{}'.format(view.__module__, view.__name__)
```
Line 267
```
#good_referrer = f'https://{request.host}/'
good_referrer = 'https://{}/'.format(request.host)
```

---
Error report:
This is an error that occurs in python versions below 3.6 where f-strings are used instead of format()
